static/app/bootstrap/exportGlobals.tsx

```diff
@@ -54,7 +56,77 @@ globals.SentryApp = {
   Modal: require('app/actionCreators/modal'),
 };
 
-// Make globals available on the window object
-Object.keys(globals).forEach(name => (window[name] = globals[name]));
+/**
+ * Wrap export so that we can track usage of these globals to determine how we want to handle deprecatation.
+ * These are sent to Sentry install, which then checks to see if SENTRY_BEACON is enabled
+ * in order to make a request to the SaaS beacon.
+ */
+let _beaconComponents: {component: string; stack: string}[] = [];
+const makeBeaconRequest = throttle(
+  async () => {
+    const api = new Client();
+
+    const components = _beaconComponents;
+    _beaconComponents = [];
+    try {
+      await api.requestPromise('/api/0/internal/beacon/', {
+        method: 'POST',
+        data: {
+          // Limit to first 20 components... if there are more than 20, then something
+          // is probably wrong.
+          batch_data: components.slice(0, 20).map(component => ({
+            description: 'SentryApp',
+            ...component,
+          })),
+        },
+      });
+    } catch (e) {
+      // Delicious failure.
+    }
+  },
+  5000,
+  {trailing: true, leading: false}
+);
+
+[
+  [SentryApp, globals.SentryApp],
+  [globals, window],
+].forEach(([obj, parent]) => {
+  const properties = Object.fromEntries(
+    Object.entries(obj).map(([key, value]) => {
+      return [
+        key,
+        {
+          configurable: false,
+          enumerable: false,
+          get() {
+            try {
+              const stack = new Error().stack;
+              // Split stack by lines and filter out empty strings
+              const stackArr = stack?.split('\n').filter(s => !!s) || [];
+              // There's an issue with Firefox where this getter for jQuery gets called many times (> 100)
+              // The stacktrace doesn't show it being called outside of this block either.
+              // And this works fine in Chrome...
+              if (key !== 'SentryApp' && stackArr.length > 1) {
```

Comment:
Thoughts on this @markstory?

Reply:
Looks good. I checked locally and beacon requests were quiet.